### PR TITLE
Sync: Start syncing Plugin Actions

### PR DESCRIPTION
**Pull request by enejb**
We want to display plugins from the .com side api endpoint. 
And we are missing info regarding the plugin action links. This PR starts syncing the new action links. 

#### Changes proposed in this Pull Request:
* 

#### Testing instructions:

* Do the tests pass? 
* test the get plugin api endpiong /sites/SITE_ID/plugin 
Do you see the actions links for jetpack ?
* Activate a plugin like Woo do you see the action links being synced on the .com side?

### CODE REVIEW

File: sync/class.jetpack-sync-functions.php [hunk 1]
```diff
@@ -274,6 +274,53 @@ public static function get_plugins() {
 		return apply_filters( 'all_plugins', get_plugins() );
 	}
 
+	/**
+	 * Get custom action link tags that the plugin is using
+	 * Ref: https://codex.wordpress.org/Plugin_API/Filter_Reference/plugin_action_links_(plugin_file_name)
+	 * @return array of plugin action links (key: link name value: url)
+	 */
+	public static function get_plugins_action_links( $plugin_file_singular = null ) {
```
Comment (lezama) [inline sync/class.jetpack-sync-functions.php]: this could be a little bit expensive to calculate, should we add a cache?

Comment (enejb) [reply sync/class.jetpack-sync-functions.php]: We don't have one now. And there is already a cache no? We only check for changes every minute. How long do you think the cache should be in place there?

Comment (lezama) [reply sync/class.jetpack-sync-functions.php]: maybe daily?

Comment (enejb) [reply sync/class.jetpack-sync-functions.php]: Good idea I will invalidate it on every plugin update and plugin activation.

**Pull request merged**